src/templates/course.js

```diff
@@ -0,0 +1,222 @@
+  let courseModuleVersion = courseVersion ? courseVersion : `latest`
+  const [visited, markVisited] = useLocalStorage(
+    dirname, `${courseModuleVersion}${location.pathname}`, false
+  )
+
+  return (
+    <DocLayout
+      title={pageContext.seo}
+      location={location}
+      gitRemote={gitRemote}
+      currentPage={location.pathname}
+      pages={parliamentNavigation.pages}
+      sideNav={
+        <SiteMenu
+          currentPage={location.pathname}
+          gitRemote={gitRemote}
+          pages={parliamentNavigation.pages}
+          CustomNavComponent={CourseNav}
+        />
+      }
+      rightRail={
+        <aside
+          css={css`
+            position: fixed;
+            top: var(--spectrum-global-dimension-size-800);
+            height: 100%;
+          `}
+        >
+          <PageActions
+            gitRemote={gitRemote}
+            relativePath={relativePath}
+            tableOfContents={tableOfContents}
+            timeToRead={timeToRead}
+          />
+          Powered by{" "}
+          <Link href="https://developers.corp.adobe.com/parliament-docs/README.md">
+            Parliament
+          </Link>
+        </aside>
+      }
+    >
+      <div
+        css={css`
+          float: right;
+          z-index: 100;
+        `}
+      >
+        <Flex alignItems="center">
+          { visited &&
+            <Flex alignItems="center">
+              <Checkmark />{" "}You have already read this module!
+            </Flex>
+          }
+
+          {gitRemote !== null ? (
+            <ActionButtons
+              gitUrl={`${gitRemote.protocol}://${gitRemote.resource}/${gitRemote.full_name}`}
+              filePath={relativePath}
+              branch={gitRemote.ref}
+            />
+          ) : (
+            ""
+          )}
+        </Flex>
+      </div>
+
+      <RenderMdx>{body}</RenderMdx>
+
+      <Flex
+        direction="column"
+        justifyContent="space-between"
+        gap="size-100"
+        marginTop="size-800"
+        marginBottom="size-400"
+      >
+        <View>
+          <NextPrev markProgression={markVisited} nextPage={nextPage} previousPage={previousPage} />
+        </View>
+        <View>
+          <Contributors
+            href={`${gitRemote.protocol}://${gitRemote.resource}/${gitRemote.full_name}/blob/${gitRemote.ref}/${relativePath}`}
+            contributors={contributors}
+            date={
+              contributors[0]
+                ? new Date(contributors[0].date).toLocaleDateString()
+                : new Date().toLocaleDateString()
+            }
+          />
+        </View>
+      </Flex>
+    </DocLayout>
+  )
+}
+
+export default CoursesTemplate
+
+export const query = graphql`
+  query CoursesTemplateQuery($id: String!) {
+    file(id: { eq: $id }) {
+      id
+      name
+      absolutePath
+      childMdx {
+        body
+        tableOfContents(maxDepth: 3)
+        timeToRead
+        frontmatter {
+          courseVersion
```

Comment:
The graphql DB is weird and needs to know about all the possible `frontmatter` at build time. So if you try to query for something that's not in a markdown like our other projects it will fail. That is why we have the `src/markdown-pages/do-not-delete.md` file to add in dummy frontmatter.

Reply:
Fixed with: https://github.com/adobe/parliament-client-template/commit/09930d5af4d6ca52a6f0414d10b6c63a81321dda

Reply:
thanks!